bitstore/controllers.py

```diff
@@ -160,3 +164,37 @@ def info(auth_token):
     except Exception as exception:
         logging.exception('Bad request (info)')
         return Response(status=400)
+
+
+def presign(auth_token, url, ownerid=None):
+    """Generates S3 presigned URLs if necessary
+    :param auth_token: authentication token from auth
+    :param ownerid: ownerid for dataset
+    :param url: url to check for sigend URL
+    """
+    s3 = get_s3_client()
+    try:
+        needs_signed_url = requests.get(url)
+        if needs_signed_url.status_code != 403:
+            return json.dumps({'url': url})
+        # Verify client, deny access if not verified
+        if ownerid is None:
+            return Response(status=400)
+        if not services.verify(auth_token, ownerid):
+            return Response(status=401)
+        parsed_url = requests.utils.urlparse(url)
+        bucket = parsed_url.netloc
+        key = parsed_url.path
+        if key.startswith('/'):
+            key = key[1:]
+
+        signed_url = s3.generate_presigned_url(
```

Comment:
So - basically a logged in user can request to pre-sign any file on the package store and gain access to it?
Need to find a better mechanism to make sure that the file actually belongs to the user.